awx/main/scheduler/task_manager.py

```diff
@@ -302,6 +304,8 @@ def post_commit():
         connection.on_commit(post_commit)
 
     def process_running_tasks(self, running_tasks):
+        # update remaining capacity on instances
+        self.instances_by_hostname = Instance.get_remaining_capacity(self.instances_by_hostname, running_tasks)
```

Comment:
I don't like the location of this, specifically, I think the call should happen earlier next to more relevant code.

Maybe I get where you're coming from, in that, you want to only consider running tasks for capacity usage.

I would outright pass `all_sorted_tasks` into `after_lock_init` and then you could run it at any point after you created `self.instances_by_hostname`.

Reply:
ok I can do that

Reply:
done